<?xml version="1.0" encoding="UTF-8"?>
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and links that are referenced are defined
-->
<robot name="two_joint_arm">
    <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
    <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
    <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
    <!--CHAINS: When a chain is specified, all the links counting from the base_link to the tip_link are included in the group. Additionally, all the joints that are parents to included links are also included. When a chain is specified, the Joints link is not included because all joints are implicitly included-->
    <!--SUBGROUPS: Groups can also be formed by referencing other groups-->
    <group name="arm_group">
        <joint name="joint1"/>
        <joint name="joint2"/>
        <!--joint name="joint2"/-->
    </group>
    <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
    <group_state name="home" group="arm_group">
        <joint name="joint1" value="0"/>
        <joint name="joint2" value="0"/>
        <!--joint name="joint2" value="0"/-->
    </group_state>
    <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
    <!-- <end_effector name="eef" parent_link="wrist" group="arm_group"/> --> <!-- Let's omit the end-effector group for now, just define the group itself -->

    <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
    <virtual_joint name="virtual_joint" type="fixed" parent_frame="base" child_link="base"/> <!-- Changed parent_frame to 'base' link as it exists -->

    <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between pairs of links. -->
    <disable_collisions link1="base" link2="shoulder" reason="Adjacent"/>
    <disable_collisions link1="elbow" link2="shoulder" reason="Adjacent"/>
    <!--disable_collisions link1="elbow" link2="wrist" reason="Adjacent"/-->
</robot> 